src/scene/renderer/renderer.js

```diff
@@ -523,15 +534,12 @@ class Renderer {
         const skinTime = now();
         // #endif
 
-        const count = drawCalls.length;
-        for (let i = 0; i < count; i++) {
-            const drawCall = drawCalls[i];
-            if (drawCall.visibleThisFrame) {
-                const skin = drawCall.skinInstance;
-                if (skin && skin._dirty) {
-                    skin.updateMatrixPalette(drawCall.node, _skinUpdateIndex);
-                    skin._dirty = false;
-                }
+        for (const drawCall of drawCalls) {
```

Comment:
How does `Set` iteration transpile in Babel? Also, it's not obvious that `drawCalls` is a `Set`. I had to jump around the code a bit to find out. Might be useful to add a JSDoc block here.

Reply:
yep adding JSDocs for these. This is the generated code:

```
for (var _iterator = _createForOfIteratorHelperLoose(drawCalls), _step; !(_step = _iterator()).done;) {
	      var drawCall = _step.value;
               ...
}
```